src/main/java/com/gtnewhorizon/gtnhlib/reflect/Fields.java

```diff
@@ -0,0 +1,391 @@
+         * @see ClassFields#getField(LookupType, String, Class)
+         */
+        public Field<Character> getCharField(@Nonnull LookupType strategy, @Nonnull String name) {
+            java.lang.reflect.Field javaField =
+                    getCheckedFieldImpl(Objects.requireNonNull(strategy), Objects.requireNonNull(name), char.class);
+            if (javaField == null) {
+                return null;
+            }
+            return new Field<>(javaField, Character.class);
+        }
+
+        /**
+         * Looks up a primitive field in the class.
+         * @see ClassFields#getField(LookupType, String, Class)
+         */
+        public Field<Float> getFloatField(@Nonnull LookupType strategy, @Nonnull String name) {
+            java.lang.reflect.Field javaField =
+                    getCheckedFieldImpl(Objects.requireNonNull(strategy), Objects.requireNonNull(name), float.class);
+            if (javaField == null) {
+                return null;
+            }
+            return new Field<>(javaField, Float.class);
+        }
+
+        /**
+         * Looks up a primitive field in the class.
+         * @see ClassFields#getField(LookupType, String, Class)
+         */
+        public Field<Double> getDoubleField(@Nonnull LookupType strategy, @Nonnull String name) {
+            java.lang.reflect.Field javaField =
+                    getCheckedFieldImpl(Objects.requireNonNull(strategy), Objects.requireNonNull(name), double.class);
+            if (javaField == null) {
+                return null;
+            }
+            return new Field<>(javaField, Double.class);
+        }
+
+        public class Field<F> {
+            public final java.lang.reflect.Field javaField;
+            public final Class<F> accessType;
+            public final boolean isPrimitive, isStatic, isFinal;
+            private final Function<C, F> getAccessor;
+            private final BiConsumer<C, F> setAccessor;
+
+            private Field(java.lang.reflect.Field javaField, Class<F> accessType) {
+                javaField.setAccessible(true);
+                this.javaField = javaField;
+                this.accessType = accessType;
+                this.isPrimitive = javaField.getType().isPrimitive();
+                this.isStatic = Modifier.isStatic(javaField.getModifiers());
+                this.isFinal = Modifier.isFinal(javaField.getModifiers());
+                try {
+                    // Generate getter
+                    final MethodHandle getterHandle = mhLookup.unreflectGetter(this.javaField);
+                    if (this.isStatic) {
+                        final MethodHandle exactGetterHandle = getterHandle.asType(MethodType.methodType(Object.class));
+                        this.getAccessor = obj -> {
+                            try {
+                                return (F) exactGetterHandle.invokeExact();
+                            } catch (Throwable e) {
+                                Throwables.propagate(e);
```

Comment:
`throw Throwables.propagate(e);`, c.f. https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate

Reply:
Good catch, I learned about this from forge code and it seems they are not throwing it in many places where they should :P